ui/desktop/src/hooks/useChatStream.ts

```diff
@@ -166,17 +183,69 @@ export function useChatStream({
     setNotifications((prev) => [...prev, notification]);
   }, []);
 
+  const stopPowerSaveBlocker = useCallback(() => {
+    try {
+      window.electron.stopPowerSaveBlocker();
+    } catch (error) {
+      console.error('Failed to stop power save blocker:', error);
+    }
+
+    if (powerSaveTimeoutId) {
+      window.clearTimeout(powerSaveTimeoutId);
+      setPowerSaveTimeoutId(null);
+    }
+  }, [powerSaveTimeoutId]);
+
   const onFinish = useCallback(
     (error?: string): void => {
+      stopPowerSaveBlocker();
+
       if (error) {
         setSessionLoadError(error);
       }
+
+      const timeSinceLastInteraction = Date.now() - lastInteractionTime;
+
+      if (timeSinceLastInteraction > 60000) {
+        try {
+          window.electron.showNotification({
+            title: 'Goose finished the task',
+            body: 'Click here for details',
+          });
+        } catch (error) {
+          console.error('Failed to show notification:', error);
+        }
+      } else {
+        console.log('useChatStream: Not showing notification (task took less than 2 seconds)');
+      }
+
+      const isNewSession = sessionId && sessionId.match(/^\d{8}_\d{6}$/);
+      if (isNewSession) {
+        console.log(
+          'useChatStream: Message stream finished for new session, emitting message-stream-finished event'
+        );
+        window.dispatchEvent(new CustomEvent('message-stream-finished'));
+      }
+
       setChatState(ChatState.Idle);
       onStreamFinish();
     },
-    [onStreamFinish]
+    [onStreamFinish, stopPowerSaveBlocker, lastInteractionTime, sessionId]
   );
 
+  useEffect(() => {
+    return () => {
+      if (powerSaveTimeoutId) {
+        window.clearTimeout(powerSaveTimeoutId);
+      }
+      try {
+        window.electron.stopPowerSaveBlocker();
+      } catch (error) {
+        console.error('Failed to stop power save blocker during cleanup:', error);
+      }
+    };
+  }, [powerSaveTimeoutId]);
```

Comment:
doesn't this immediately run when powerSaveTimeoutId gets set and therefore immediately clears the timer and the power saver blocker?